### ISSUE TITLE
Game crashes opening the VAB

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#193]
I scheduled an Apollo suborbital flight and ended my turn.  When I opened the VAB next turn and clicked on the mission to assign hardware for it, the game crashed with the following message:

free(): invalid next size (normal)
Aborted (core dumped)

**Comment by rnyoakum** [comment]
This hasn't happened to any of my saves, but I haven't had time to actually play it since issue182 got fixed. My saves only go so far as planning a Gemini suborbital. Can the bug be reliably replicated? If so (or if you haven't time to test), could you send me a copy of the save game so I can try to track this down?

Also, did you assemble a Gemini launch or another launch vehicle which used the large casing? The issue182 fix should work fine (and seems to work perfectly for me) but if it creates some other problem, or was hiding yet another bug, I've been busy and haven't played several turns beyond my test save.

In case it's related to any of the other memory bugs that happen, I'm curious if you have problems with your saves like I do? Since I started playing (before joining the project), every time I save I've gotten a segmentation fault, and the program frequently crashes when I load a game.

**Comment by peyre** [comment]
Ok, I deleted the game folder, cloned it fresh from github and recompiled the game.  Then I started a new game, and when I started a turn with an Apollo mission scheduled, it crashed with the same error: 

free(): invalid next size (normal)
Aborted (core dumped)

I've attached my autosave.

    On Saturday, November 10, 2018, 9:30:50 PM PST, Ryan Yoakum <notifications@github.com> wrote:  
 
 
This hasn't happened to any of my saves, but I haven't had time to actually play it since issue182 got fixed. My saves only go so far as planning a Gemini suborbital. Can the bug be reliably replicated? If so (or if you haven't time to test), could you send me a copy of the save game so I can try to track this down?

Also, did you assemble a Gemini launch or another launch vehicle which used the large casing? The issue182 fix should work fine (and seems to work perfectly for me) but if it creates some other problem, or was hiding yet another bug, I've been busy and haven't played several turns beyond my test save.

In case it's related to any of the other memory bugs that happen, I'm curious if you have problems with your saves like I do? Since I started playing (before joining the project), every time I save I've gotten a segmentation fault, and the program frequently crashes when I load a game.

**Comment by rnyoakum** [comment]
When I made the fix to issue182 where the VAB crashed, I accidently made the cutoff point for how much filler is used on the large payload casing (used by all multi-person capsules) one pixel too big. This caused it to draw past the buffer boundaries. When I tested, the Gemini capsule image is small enough the entire casing filler image was used, so the cutoff was never needed. With the new fix (8ddb0240da33e7685bdcefd172ba47ebe8d26558) I'm no longer getting the problem on the save you provided.

I'm very sorry about that, and I'll wait until you confirm the bug is handled before closing this issue.
